### PR TITLE
Key Vault: refactored keys and secrets readme based on ux feedback

### LINKED ISSUE #8211: Refactor keys and secrets README with ux design review feedback
- [ ] link to class docs
- [ ] make all examples basically runnable, show imports
- [ ] call out azure-identity as prereq
- [ ] vault url is "DNS Name" in portal, that's confusing

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #8211

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-keys/README.md [hunk 1]
```diff
@@ -165,42 +182,68 @@ print(ec_key.key_type)
...
+
+credential = DefaultAzureCredential()
+
+key_client = KeyClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+
 # Clients may specify additional application-specific metadata in the form of tags.
 tags = {"foo": "updated tag"}
 
-updated_key_properties = key_client.update_key_properties("key-name", tags=tags)
+updated_key = key_client.update_key_properties("key-name", tags=tags)
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-keys/README.md]: Everyone who read this thought we were trying to teach them that updateable properties are all tags. So I'm thinking we'd better show e.g. disabling a key instead.

Comment (iscai-msft) [reply sdk/keyvault/azure-keyvault-keys/README.md]: switched the keys one to just toggle enable, while the secrets one both disables the secret and updates content-type. Lmk what you think!

**Pull request merged**